crates/ruff_db/src/files.rs

```diff
@@ -27,6 +27,12 @@ pub fn system_path_to_file(db: &dyn Db, path: impl AsRef<SystemPath>) -> Option<
     file.exists(db).then_some(file)
 }
 
+/// Interns a virtual file system path and returns a salsa [`File`] ingredient.
+#[inline]
+pub fn system_virtual_path_to_file(db: &dyn Db, path: impl AsRef<SystemVirtualPath>) -> File {
+    db.files().system_virtual(db, path.as_ref())
+}
+
```

Comment:
I would remove this method for now and instead add a `Files::add_virtual_file(path)` method (we'll want a `remove_virtual_file` long term). The reason why I think we don't need a `system_virtual_path_to_file` function is because we should never really have the need to go from path to file. Instead, the code that creates the virtual file should hold on to it (the LSP should store the `File` directly).